mm2src/coins/eth.rs

```diff
@@ -4842,52 +4909,136 @@ impl EthCoin {
-                    match GasStationData::get_gas_price(url, coin.gas_station_decimals, coin.gas_station_policy.clone())
-                        .compat()
-                        .await
-                    {
-                        Ok(from_station) => Some(increase_by_percent_one_gwei(from_station, GAS_PRICE_PERCENT)),
-                        Err(e) => {
-                            error!("Error {} on request to gas station url {}", e, url);
-                            None
-                        },
-                    }
-                },
-                None => None,
-            };
-
-            let eth_gas_price = match coin.gas_price().await {
-                Ok(eth_gas) => Some(eth_gas),
-                Err(e) => {
-                    error!("Error {} on eth_gasPrice request", e);
-                    None
-                },
-            };
-
-            let eth_fee_history_price = match coin.eth_fee_history(U256::from(1u64), BlockNumber::Latest, &[]).await {
-                Ok(res) => res
-                    .base_fee_per_gas
-                    .first()
-                    .map(|val| increase_by_percent_one_gwei(*val, BASE_BLOCK_FEE_DIFF_PCT)),
-                Err(e) => {
-                    debug!("Error {} on eth_feeHistory request", e);
-                    None
-                },
-            };
+            let eth_gas_price_fut = async {
+                match coin.gas_price().await {
+                    Ok(eth_gas) => Some(eth_gas),
+                    Err(e) => {
+                        error!("Error {} on eth_gasPrice request", e);
+                        None
+                    },
+                }
+            }
+            .boxed();
+
+            let eth_fee_history_price_fut = async {
+                match coin.eth_fee_history(U256::from(1u64), BlockNumber::Latest, &[]).await {
+                    Ok(res) => res
+                        .base_fee_per_gas
+                        .first()
+                        .map(|val| increase_by_percent_one_gwei(*val, BASE_BLOCK_FEE_DIFF_PCT)),
+                    Err(e) => {
+                        debug!("Error {} on eth_feeHistory request", e);
+                        None
+                    },
+                }
+            }
+            .boxed();
 
+            let (eth_gas_price, eth_fee_history_price) = join(eth_gas_price_fut, eth_fee_history_price_fut).await;
             // on editions < 2021 the compiler will resolve array.into_iter() as (&array).into_iter()
             // https://doc.rust-lang.org/edition-guide/rust-2021/IntoIterator-for-arrays.html#details
-            IntoIterator::into_iter([gas_station_price, eth_gas_price, eth_fee_history_price])
+            IntoIterator::into_iter([eth_gas_price, eth_fee_history_price])
                 .flatten()
                 .max()
                 .or_mm_err(|| Web3RpcError::Internal("All requests failed".into()))
         };
         Box::new(fut.boxed().compat())
     }
```

Comment:
Again, this can be much simpler if made async.

Reply:
done in [531f1af](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2051/commits/531f1afb6832974583ae7110575769b952393e77)